src/objective-c/GRPCClient/private/GRPCHost.m

```diff
@@ -127,9 +205,14 @@ - (GRPCChannel *)channel {
         args[@GRPC_SSL_TARGET_NAME_OVERRIDE_ARG] = _hostNameOverride;
       }
 
-      _channel = [GRPCChannel secureChannelWithHost:_address
-                                 pathToCertificates:_pathToCertificates
-                                        channelArgs:args];
+      @synchronized(self) {
+        if (_channelCreds == nil) {
+          [self setSSLCredentials:nil withPrivateKey:nil withCertChain:nil error:nil];
+        }
+        _channel = [GRPCChannel secureChannelWithHost:_address
+                                          credentials:_channelCreds
+                                          channelArgs:args];
+      }
```

Comment:
My feeling is that doing (2) by adding a `GRPCChannelCredentials` class that creates the C object and manages its life (and has `[GRPCChannelCredentials defaultCredentials]` and whatnot) is The Right Way (TM). But it might not be worth the hassle if we're just improving readability with it. So I'm happy with (4) too.